azure-iot-device/azure/iot/device/common/transport/mqtt/pipeline_ops.py

```diff
@@ -0,0 +1,342 @@
+        Constructor to instantiate a `Mqtt_SetConnectionArgs` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.client_id = client_id
+        self.hostname = hostname
+        self.username = username
+        self.device_id = device_id
+        self.module_id = module_id
+        self.ca_cert = ca_cert
+
+    pass
+
+
+class Mqtt_Publish(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to publish a specific payload on a specific topic using the Mqtt protocol.
+
+    This operation is in the group of "Mqtt" operations becauase its attributes are very specific to the Mqtt protocol.
+    """
+
+    def __init__(self, topic=None, payload=None, qos=1, callback=None):
+        """
+        Constructor to instantiate a `Mqtt_Publish` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.topic = topic
+        self.qos = qos
+        self.payload = payload
+
+
+class Mqtt_Subscribe(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to subscribe to a specific Mqtt topic using the Mqtt protocol.
+
+    This operation is in the group of "Mqtt" operations becauase its attributes are very specific to the Mqtt protocol.
+    """
+
+    def __init__(self, topic=None, qos=1, callback=None):
+        """
+        Constructor to instantiate a `Mqtt_Subscribe` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.topic = topic
+        self.qos = qos
+
+
+class Mqtt_Unsubscribe(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to unsubscribe from a specific Mqtt topic using the Mqtt protocol.
+
+    This operation is in the group of "Mqtt" operations becauase its attributes are very specific to the Mqtt protocol.
+    """
+
+    def __init__(self, topic=None, callback=None):
+        """
+        Constructor to instantiate a `Mqtt_Unsubscribe` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
```

Comment:
>:param callback Functio [](start = 8, length = 23)

Here and everywhere the initializers are missing parameters in the docs.

Also nit, but these aren't constructors, they're initializers. The constructor is something else (that we never touch)